py/test/selenium/webdriver/firefox/ff_profile_tests.py

```diff
@@ -171,3 +171,52 @@ def test_autodetect_proxy_is_set_in_profile():
 
     profile.set_proxy(proxy)
     assert profile.default_preferences["network.proxy.type"] == ProxyType.AUTODETECT['ff_value']
+
+
+def test_add_extension_web_extension_with_id(capabilities, webserver):
+    current_directory = os.path.dirname(os.path.realpath(__file__))
+    root_directory = os.path.join(current_directory, '..', '..', '..', '..', '..')
+    # TODO: This file should probably live in a common directory.
+    extension_path = os.path.join(root_directory, 'javascript', 'node', 'selenium-webdriver',
+                                  'lib', 'test', 'data', 'firefox', 'webextension.xpi')
+
+    profile = FirefoxProfile()
+    profile.add_extension(extension_path)
+
+    driver = Firefox(capabilities=capabilities, firefox_profile=profile)
+    profile_path = driver.firefox_profile.path
+    extension_path_in_profile = os.path.join(profile_path, 'extensions', 'webextensions-selenium-example@example.com')
+    assert os.path.exists(extension_path_in_profile)
+    driver.get(webserver.where_is('simpleTest.html'))
+    driver.find_element_by_id('webextensions-selenium-example')
+    driver.quit()
+
+
+def test_add_extension_web_extension_without_id(capabilities, webserver):
```

Comment:
This test is failing for me.

Reply:
Is this due to the reasons you mentioned in https://github.com/SeleniumHQ/selenium/pull/6463/files#r221901851 or some other reason?

Reply:
The web extension doesn't appear to be copied to the profile path at all.

Reply:
As this is passing in Travis, let's go ahead and merge. It was the only thing that was making me uncertain about merging. It's failing locally for me, but I'm guessing it's related to the version of Firefox.